Out of bounds access on echoarea entering

Got

2020/03/08 12:22:22 gossipEd-linux/amd64 2.0-dev started
panic: runtime error: slice bounds out of range [recovered]
        panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/rivo/tview.(*Application).Run.func1(0xc0000a6000)
        /home/new/go/pkg/mod/github.com/rivo/tview@v0.0.0-20200204110323-ae3d8cac5e4b/application.go:160 +0x82
panic(0x60b4a0, 0x7ca080)
        /usr/lib/go-1.11/src/runtime/panic.go:513 +0x1b9
github.com/askovpen/gossiped/pkg/msgapi.(*Squish).GetMsg(0xc000496000, 0x7f1400000001, 0x0, 0x0, 0x0)
        /home/new/src/gossiped/pkg/msgapi/squish.go:188 +0xe61
github.com/askovpen/gossiped/pkg/ui.(*App).ViewMsg(0xc0002d8620, 0xa3, 0x0, 0x0, 0x2, 0xc0004714c0, 0x11, 0x2)
        /home/new/src/gossiped/pkg/ui/viewmsg.go:16 +0x7b
github.com/askovpen/gossiped/pkg/ui.(*App).onSelected(0xc0002d8620, 0xa4, 0x0)
        /home/new/src/gossiped/pkg/ui/arealist.go:122 +0x3c4
github.com/askovpen/gossiped/pkg/ui.(*App).AreaList.func3(0xc0004bc3c0, 0xc000088001)
        /home/new/src/gossiped/pkg/ui/arealist.go:87 +0x204
github.com/rivo/tview.(*Box).WrapInputHandler.func1(0xc0004bc3c0, 0xc0000881d0)
        /home/new/go/pkg/mod/github.com/rivo/tview@v0.0.0-20200204110323-ae3d8cac5e4b/box.go:157 +0x75
github.com/rivo/tview.(*Application).Run(0xc0000a6000, 0x0, 0x0)
        /home/new/go/pkg/mod/github.com/rivo/tview@v0.0.0-20200204110323-ae3d8cac5e4b/application.go:245 +0x3d3
github.com/askovpen/gossiped/pkg/ui.(*App).Run(0xc0002d8620, 0x1, 0x1)
        /home/new/src/gossiped/pkg/ui/app.go:39 +0x56
main.main()
        /home/new/src/gossiped/gossiped.go:73 +0x468
zsh: exit 2     gossiped

on entering z2daily

--- a/pkg/msgapi/squish.go
+++ b/pkg/msgapi/squish.go
@@ -185,11 +185,13 @@ func (s *Squish) GetMsg(position uint32) (*Message, error) {
 			rm.Replies = append(rm.Replies, s.getOffsetByNum(reply))
 		}
 	}
-	kla := strings.Split(rm.Body[1:sqdh.CLen], "\x01")
-	for i := range kla {
-		kla[i] = strings.Trim(kla[i], "\x00")
+	if sqdh.CLen > 0 {
+		kla := strings.Split(rm.Body[1:sqdh.CLen], "\x01")
+		for i := range kla {
+			kla[i] = strings.Trim(kla[i], "\x00")
+		}
+		rm.Body = "\x01" + strings.Join(kla, "\x0d\x01") + "\x0d" + rm.Body[sqdh.CLen:]
 	}
-	rm.Body = "\x01" + strings.Join(kla, "\x0d\x01") + "\x0d" + rm.Body[sqdh.CLen:]
 	if strings.Contains(rm.Body, "\x00") {
 		rm.Body = rm.Body[0:strings.Index(rm.Body, "\x00")]
 	}

